File: src/main/java/com/github/timo_reymann/ansible_vault_integration/execution/runnable/AnsibleVaultRunnable.java
package com.github.timo_reymann.ansible_vault_integration.execution.runnable;

import org.jetbrains.annotations.NotNull;

/**
 * Abstraction for ansible vault task
 */
public interface AnsibleVaultRunnable {
    /***
     * Execute task using ansible vault
     * @throws Exception Exceptions during execution
     */
    void run() throws Exception;

    /**
     * Get message for success
     *
     * @return String
     */
    @NotNull
    String getSuccessMessage();
}

File: src/main/kotlin/de/timo_reymann/ansible_vault_integration/runnable/AnsibleVaultRunnable.kt
package de.timo_reymann.ansible_vault_integration.runnable

enum class VaultRunnableType {
    ENCRYPT,
    DECRYPT;

    override fun toString(): String {
        return when (this) {
            ENCRYPT -> "Vault"
            DECRYPT -> "Unvault"
        }
    }

    fun stringify(fileName: String): String {
        return "$this $fileName"
    }
}

enum class VaultRunnableMode {
    INLINE,
    FILE
}

/**
 * Abstraction for ansible vault task
 */
interface AnsibleVaultRunnable {
    /***
     * Execute task using ansible vault
     * @throws Exception Exceptions during execution
     */
    @Throws(Exception::class)
    fun run()

    /**
     * File name to display
     */
    val fileName: String

    /**
     * Type of the runnable
     */
    val type: VaultRunnableType

    /**
     * Specifies if the action was run for entire file(s) or a single value
     */
    val mode : VaultRunnableMode
}
